src/notification-banner/private/_Banner.jsx

```diff
@@ -0,0 +1,208 @@
+      return null;
+    case AlertTypes.INFO:
+      return (<span className={cx('icon')}><IconInformation /></span>);
+    case AlertTypes.SUCCESS:
+      return (<span className={cx('icon')}><IconSuccess /></span>);
+    case AlertTypes.CUSTOM:
+      return (<span className={cx('icon')}>{customIcon}</span>);
+    default:
+      return null;
+  }
+};
+
+const Alert = ({
+  action,
+  children,
+  customIcon,
+  customColorClass,
+  onDismiss,
+  title,
+  type,
+  ...customProps
+}) => {
+  const theme = React.useContext(ThemeContext);
+
+  const [isNarrow, setIsNarrow] = useState();
+
+  const defaultTitle = type === AlertTypes.CUSTOM ? '' : <FormattedMessage id={`Terra.alert.${type}`} />;
+  const alertClassNames = classNames(
+    cx(
+      'alert-base',
+      type,
+      { narrow: isNarrow },
+      { wide: !isNarrow },
+      theme.className,
+    ),
+    customProps.className,
+    { [`${customColorClass}`]: customColorClass && type === AlertTypes.CUSTOM },
+  );
+
+  const bodyClassNameForParent = cx(
+    'body',
+    { 'body-std': !isNarrow || (isNarrow && !onDismiss && !action) },
+    { 'body-narrow': isNarrow && (onDismiss || action) },
+  );
+
+  let dismissButton;
+  if (onDismiss) {
+    dismissButton = (
+      <FormattedMessage id="Terra.alert.dismiss">
+        {buttonText => (
+          <Button
+            text={buttonText}
+            onClick={onDismiss}
+            aria-labelledby="#alert"
+            // aria-describedBy={`Dismiss ${type} Notification Banner.`}
+            // name={`Click to dismiss the ${defaultTitle} Notification Banner.`}
+          />
+        )}
+      </FormattedMessage>
+    );
+  }
+
+  let actionsSection;
+  if (onDismiss || action) {
+    const actionsClassName = cx('actions', { 'actions-custom': type === AlertTypes.CUSTOM });
+    actionsSection = (
+      <div className={actionsClassName}>
+        {action}
+        {dismissButton}
+      </div>
+    );
+  }
+
+  const alertSectionClassName = cx('section', { 'section-custom': type === AlertTypes.CUSTOM });
+  const alertMessageContent = (
+    <div className={alertSectionClassName}>
+      {(title || defaultTitle) && <strong className={cx('title')}>{title || defaultTitle}</strong>}
+      {children}
+    </div>
+  );
+
+  return (
+    <ResponsiveElement
+      onChange={(value) => {
+        const showNarrowAlert = value === 'tiny';
+        if (showNarrowAlert !== isNarrow) {
+          setIsNarrow(showNarrowAlert);
+        }
+      }}
+    >
+      <div
+        // role="alert"
+        {...customProps}
+        className={alertClassNames}
+      >
+        <div className={bodyClassNameForParent}>
+          {getAlertIcon(type, customIcon)}
+          {alertMessageContent}
+        </div>
+        {actionsSection}
+      </div>
+    </ResponsiveElement>
+
+  );
+};
+
+Alert.propTypes = propTypes;
+Alert.defaultProps = defaultProps;
+Alert.Opts = {};
+Alert.Opts.Types = AlertTypes;
+
+export default Alert;
```

Comment:
Reminder to update the component name

Reply:
Doh, this is for testing isn't it?

Reply:
Yeah this is added in create the test pages for the Accessibility team as the determine the expected behavior. We may or may not need to pull this forward now. Still waiting on latest UX designs as well.